lib/rack/server.rb

```diff
@@ -359,7 +359,7 @@ def daemonize_app
 
       def write_pid
         ::File.open(options[:pid], ::File::CREAT | ::File::EXCL | ::File::WRONLY ){ |f| f.write("#{Process.pid}") }
-        at_exit { ::File.delete(options[:pid]) if ::File.exist?(options[:pid]) }
+        at_exit { ::File.delete(options[:pid]) rescue Errno::ENOENT }
```

Comment:
Subtle – need to wrap in `begin … rescue … end` in this case since inline rescues don't take an exception class (this is actually rescuing StandardError and returning Errno::ENOENT). Switching to stdlib `FileUtils#rm_f` would also neatly wrap this up.